### PR TITLE
COMMERCE-12431 OrgChart View/Edit Panel - Account + Automation

### PR DESCRIPTION
**Pull request by baharturk**
Story: https://liferay.atlassian.net/browse/COMMERCE-12431

Test Plan: https://liferay.atlassian.net/browse/COMMERCE-12822

sent from: #4402

### CODE REVIEW

**Comment by baharturk** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 1]
```diff
@@ -304,6 +400,17 @@ definition {
 		Click(locator1 = "Button#EXPAND");
 	}
 
+	@summary = "This macro can be used for navigating to the Organization Management Widget entry."
+	macro searchAndGoToWidgetEntry(entryName = null) {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:404]: Be more specific in the macro name as it is focused on the Organization Management widget

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 2]
```diff
@@ -212,6 +236,63 @@ definition {
...
+		}
+		else {
+			Click(locator1 = "Icon#SIDEBAR_HEADER_ACTION_VERTICAL_ELLIPSIS_BUTTON");
+
+			Click(
+				key_menuItem = "Edit",
+				locator1 = "MenuItem#DROPDOWN_MENU_ITEM");
+		}
+
+		if (isSet(newAccountImage)) {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:266]:
```suggestion
		if (isSet(accountImage)) {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 3]
```diff
@@ -212,6 +236,63 @@ definition {
...
+			User.changeProfileImage(imageName = ${newAccountImage});
+		}
+
+		if (isSet(newAccountName)) {
+			PortletEntry.inputName(name = ${newAccountName});
+		}
+
+		if (isSet(newAccountTaxID)) {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:274]:
```suggestion
		if (isSet(accountTaxID)) {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 4]
```diff
@@ -212,6 +236,63 @@ definition {
...
+			PortletEntry.inputText(
+				fieldLabel = "Tax ID",
+				text = ${newAccountTaxID});
+		}
+
+		if (isSet(newAccountERC)) {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:280]:
```suggestion
		if (isSet(accountERC)) {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 5]
```diff
@@ -212,6 +236,63 @@ definition {
...
+			Type(
+				key_label = "External Reference Code",
+				locator1 = "TextArea#ANY_LABELED",
+				value1 = ${newAccountERC});
+		}
+
+		if (isSet(newAccountDescription)) {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:287]:
```suggestion
		if (isSet(accountDescription)) {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro [hunk 6]
```diff
@@ -585,4 +692,28 @@ definition {
 		SelectFrameTop();
 	}
 
+	@summary = "This macro can be used for asserting account information from Organization Management Chart."
+	macro viewAccountDetailsAtSidebar(accountDescription = null, accountERC = null, accountID = null, accountImage = null, accountName = null, accountTaxID = null, accountType = null) {
+		if (isSet(viewCurrentInfo)) {
+			CommerceWidget.searchAndGoToWidgetEntry(entryName = ${currentAccountName});
+
+			Click(
+				key_nodeName = ${currentAccountName},
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceWidget.macro:701]:
```suggestion
				key_nodeName = ${accountName},
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase [hunk 7]
```diff
@@ -1153,6 +1153,302 @@ definition {
...
+
+			CommerceNavigator.gotoMiniumSidebarItem(menuItem = "Account Management");
+
+			LexiconEntry.viewNoEntry(rowEntry = "Commerce Account");
+		}
+	}
+
+	@description = "COMMERCE-12824. As a user, I can edit account details from Organization Management Chart."
+	@priority = 5
+	test CanUserEditAccountsDetailsFromOrganizationManagementChart {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase:1223]:
```suggestion
	test CanUserEditAccountDetailsFromOrganizationManagementChart {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase [hunk 8]
```diff
@@ -1153,6 +1153,302 @@ definition {
...
+			CommerceNavigator.gotoMiniumSidebarItem(menuItem = "Organization Management Widget Page");
+		}
+
+		task ("Then can view current account details") {
+			var accountName = "Commerce Account";
+
+			CommerceWidget.viewAccountDetailsAtSidebar(
+				accountDescription = "Account Entry Description",
+				accountERC = ${accountERC},
+				accountID = ${accountID},
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase:1271]: these lines can be removed.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase [hunk 9]
```diff
@@ -1153,6 +1153,302 @@ definition {
...
+				newAccountTaxID = 1234);
+		}
+
+		task ("Then the edited fields are saved and shown correctly") {
+			var accountName = "New Account Name";
+
+			CommerceWidget.viewAccountDetailsAtSidebar(
+				accountDescription = "New Account Description",
+				accountERC = ${accountERC},
+				accountID = ${accountID},
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceOrganizationManagementWidget.testcase:1296]: these lines can be removed

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by baharturk** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/4451

**Pull request closed without merging**